syntax = "proto3";
package saber;

import "saber.proto";

message Transaction {
  uint32 group_id = 1;
  uint64 instance_id = 2;
  uint64 time = 3;
  uint64 session_id = 4;
}

message DataNode {
  Stat stat = 1;
  bytes data = 2;
  string path = 3;
  repeated string children = 4;
}

message DataNodeList {
  repeated DataNode nodes = 1;
}

message Session {
  uint64 session_id = 1;
  uint64 version = 2;
}

message SessionList {
  repeated Session sessions = 1;
}
